unstructured/ingest/connector/sharepoint.py

```diff
@@ -356,3 +417,158 @@ def get_ingest_docs(self):
             site_client = self.connector_config.get_site_client(site_url)
             ingest_docs = ingest_docs + self._ingest_site_docs(site_client)
         return ingest_docs
+
+
+class PermissionsConnector:
```

Comment:
Do get all the benefits of it and keep the pattern, this can be a `@dataclass` with the access token set in a `__post_init__`. This would also guarantee sensitive info isn't serialized.

Reply:
Agreed and addressed with https://github.com/Unstructured-IO/unstructured/pull/1628/commits/145452256c5f42c2907d6fad6486152bdf4669e9

Somehow, `__post_init__` did not run by itself after `__init__` even though `@dataclass` decorator was there; so I implemented it in a different way

Two questions:

How do we guarantee that sensitive info doesn't get serialized with this approach? Is it because `access_token` is not a property anymore, since it is not defined in init? 

Do we have similar concerns for the info in say `SimpleSharepointConfig`?